### PR TITLE
Clarify serialization behavior of "use cache"

### PR DESCRIPTION
**Pull request by gaearon**
Maybe not ideal wording but this should be more correct. See https://github.com/vercel/next.js/pull/72081#discussion_r1828278059.

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 27a5f95e8ef5b75232fb0398e8d6885c9f217fa0

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by gaearon** [comment]
slightly reworded.

**Review by unstubbable — APPROVED** [review]
Sounds good to me. Requested a review from @delbaoliveira for final approval.

File: docs/02-app/02-api-reference/01-directives/use-cache.mdx [hunk 1]
```diff
@@ -58,7 +58,8 @@ export async function getData() {
 ## Good to know
 
 - `use cache` is an experimental Next.js feature, and not a native React feature like [`use client`](/docs/app/api-reference/directives/use-client) or [`use server`](/docs/app/api-reference/directives/use-server).
-- The arguments (or props) passed to a cached function must be [serializable](https://react.dev/reference/rsc/use-server#serializable-parameters-and-return-values) by React. This means they can be converted to a format like JSON. Non-serializable props like functions or React elements cannot be used as arguments.
+- Any [serializable](https://react.dev/reference/rsc/use-server#serializable-parameters-and-return-values) arguments (or props) to the cached function, as well as any serializable values it reads from the parent scope, will be converted to a format like JSON and automatically become a part of the cache key.
```
Comment (delbaoliveira) [inline docs/02-app/02-api-reference/01-directives/use-cache.mdx:61]:
```suggestion
- Any [serializable](https://react.dev/reference/rsc/use-server#serializable-parameters-and-return-values) arguments (or props) passed to the cached function, as well as any serializable values it reads from the parent scope, will be converted to a format like JSON and automatically become a part of the cache key.
```

File: docs/02-app/02-api-reference/01-directives/use-cache.mdx [hunk 2]
```diff
@@ -58,7 +58,8 @@ export async function getData() {
...
+- Any non-serializable arguments, props, or closed-over values will turn into opaque references inside the cached function. For example, a cached function can take some JSX as a `children` prop and return it like `<div>{children}</div>` but it won't be able to introspect the actual `children` object. All non-serializable values will be filled in at the request time and won't become a part of the cache key.
```
Comment (delbaoliveira) [inline docs/02-app/02-api-reference/01-directives/use-cache.mdx:62]:
```suggestion
- Any non-serializable arguments, props, or closed-over values will turn into opaque references inside the cached function, and can be only passed through and not inspected nor modified. These non-serializable values will be filled in at the request time and won't become a part of the cache key.
  - For example, a cached function can take in JSX as a `children` prop and return `<div>{children}</div>`, but it won't be able to introspect the actual `children` object. 
```

**Review by delbaoliveira — CHANGES_REQUESTED** [review]
Thank you for further clarifying it Dan.

Tbh, the how `use cache` information is so essential that shouldn't live under a good to know. I'll move it out to its own in the future.

**Review by delbaoliveira — APPROVED** [review]

**Review by ijjk — APPROVED** [review]

**Pull request merged**